exercises/run-length-encoding/src/example/java/RunLengthEncoding.java

```diff
@@ -0,0 +1,47 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class RunLengthEncoding {
+    public String encode(String data) {
+        String encodedData = "";
+
+        for (int i = 0; i < data.length(); i++) {
```

Comment:
I think that's a great idea!  I was actually thinking of writing it this way after I had finished it.  Implementing it this way would increase readability a lot.